kge/job/train.py

```diff
@@ -51,21 +52,23 @@ def __init__(
         self.batch_size: int = config.get("train.batch_size")
         self.device: str = self.config.get("job.device")
         self.train_split = config.get("train.split")
-        valid_conf = config.clone()
-        valid_conf.set("job.type", "eval")
-        if self.config.get("valid.split") != "":
-            valid_conf.set("eval.split", self.config.get("valid.split"))
-        valid_conf.set("eval.trace_level", self.config.get("valid.trace_level"))
-        self.valid_job = EvaluationJob.create(
-            valid_conf, dataset, parent_job=self, model=self.model
-        )
         self.config.check("train.trace_level", ["batch", "epoch"])
         self.trace_batch: bool = self.config.get("train.trace_level") == "batch"
         self.epoch: int = 0
-        self.valid_trace: List[Dict[str, Any]] = []
         self.is_prepared = False
         self.model.train()
 
+        if config.get("job.type") == "train":
```

Comment:
I suggest to disable validation with an optional constructor argument instead of this more hacky approach.

Reply:
Only as constructor or as a config?

Reply:
In the config, we have it already. That being said, forget the constructor argument and simply set "valid.every" to 0 in the TrainingLossEvacluatinJob.

Reply:
Likewise, set train.checkpoint.every to 0.

Reply:
Hm, I just realized that I was still initializing optimizer (and loss). Optimizer creates new parameters based on model.parameters(). In this context querying job.type makes more sense. See new commit.

Reply:
You are assuming that the outside eval job does not have job.type=train (it could have). Either set this outside or add a constructor argument "initialize_for_forward_only=False" (preferred since clearer, I think).

Reply:
> You are assuming that the outside eval job does not have job.type=train (it could have). 

Huh? Isn't it one config == one job? "job.type" is either eval, train, search ..... When I pass trough the eval's config, then its job.type is "eval". I must misunderstand what you are saying here?

> Either set this outside or add a constructor argument "initialize_for_forward_only=False" (preferred since clearer, I think).

Ok can do.

Reply:
Not if the job is created via EvaluationJob.create(...). An explicit argument makes this clearer